internal/mvcc/backend/backend.go

```diff
@@ -47,6 +47,12 @@ var (
 )
 
 type Backend interface {
+	// StaleConcurrentReadTx returns a transcation for concurrent read. It does not observe
+	// the uncommitted writes happened in the pending or any future batch transcations.
+	// It does not block batch transcation.
+	StaleConcurrentReadTx() ReadTx
```

Comment:
`StaleTx() <-chan ReadTx` so it can wait for commit

Reply:
OK, not that interface exactly because it would have to broadcast to all open channels. 

Better:
```go
type Backend interface {
...
// Flushed returns a channel that closes when all pending data at the time of being called is committed.
Flushed() <-chan struct{}
StaleTx() ReadTx
...
}
```
so the fallback code can do:
```go
select {
case <-be.Flushed():
case <-done: 
  return
}
tx := be.StaleTx()
```